Firestore/core/src/firebase/firestore/remote/remote_objc_bridge.h

```diff
@@ -17,158 +17,125 @@
 #ifndef FIRESTORE_CORE_SRC_FIREBASE_FIRESTORE_REMOTE_REMOTE_OBJC_BRIDGE_H_
 #define FIRESTORE_CORE_SRC_FIREBASE_FIRESTORE_REMOTE_REMOTE_OBJC_BRIDGE_H_
 
-#if !defined(__OBJC__)
-#error "This header only supports Objective-C++"
-#endif  // !defined(__OBJC__)
-
-#import <Foundation/Foundation.h>
-
 #include <memory>
 #include <string>
+#include <utility>
 #include <vector>
 
+#include "Firestore/Protos/nanopb/google/firestore/v1/firestore.nanopb.h"
 #include "Firestore/core/src/firebase/firestore/core/database_info.h"
 #include "Firestore/core/src/firebase/firestore/local/query_data.h"
 #include "Firestore/core/src/firebase/firestore/model/snapshot_version.h"
 #include "Firestore/core/src/firebase/firestore/model/types.h"
 #include "Firestore/core/src/firebase/firestore/nanopb/byte_string.h"
+#include "Firestore/core/src/firebase/firestore/nanopb/message.h"
+#include "Firestore/core/src/firebase/firestore/remote/serializer.h"
 #include "Firestore/core/src/firebase/firestore/remote/watch_change.h"
 #include "Firestore/core/src/firebase/firestore/util/status_fwd.h"
-#include "absl/types/optional.h"
 #include "grpcpp/support/byte_buffer.h"
 
-#import "Firestore/Protos/objc/google/firestore/v1/Firestore.pbobjc.h"
-#import "Firestore/Source/Remote/FSTSerializerBeta.h"
-
 namespace firebase {
 namespace firestore {
 namespace remote {
 namespace bridge {
 
 bool IsLoggingEnabled();
 
-/**
- * This file contains operations in remote/ folder that are still delegated to
- * Objective-C: proto parsing and delegates.
- *
- * The principle is that the C++ implementation can only take Objective-C
- * objects as parameters or return them, but never instantiate them or call any
- * methods on them -- if that is necessary, it's delegated to one of the bridge
- * classes. This allows easily identifying which parts of remote/ still rely on
- * not-yet-ported code.
- */
+// TODO(varconst): remove this file?
```

Comment:
Honestly, I kind of like the current code structure, even though it diverges from other platforms. Let me know what you think.